packages/atoms/src/SearchInput.tsx

```diff
@@ -0,0 +1,96 @@
+import React, { useState } from 'react';
+import { prepareNativeStyle, useNativeStyles } from '@trezor/styles';
+import { TextInput, Pressable, View } from 'react-native';
+import { Box } from './Box';
+import { Icon } from './Icon/Icon';
+
+const MAX_LENGTH = 255;
+
+type InputProps = {
+    value: string;
+    onChange: (value: string) => void;
+    placeholder: string;
+    isClearable?: boolean;
+    isDisabled?: boolean;
+};
+
+type InputStyleProps = {
+    isFocused: boolean;
+};
+
+const inputStyle = prepareNativeStyle(utils => ({
+    flex: 1,
+    fontSize: 16,
+    color: utils.colors.gray700,
+    height: '100%',
+    marginLeft: utils.spacings.md,
+}));
+
+const clearIconStyle = prepareNativeStyle(utils => ({
+    backgroundColor: utils.colors.gray500,
+    height: 20,
+    width: 20,
+    alignItems: 'center',
+    justifyContent: 'center',
+    borderRadius: utils.borders.radii.round,
+}));
+
+const inputWrapperStyle = prepareNativeStyle<InputStyleProps>((utils, { isFocused }) => ({
+    flexDirection: 'row',
+    justifyContent: 'space-between',
+    alignItems: 'center',
+    height: 48,
+    borderWidth: 1,
+    borderColor: utils.colors.gray200,
+    backgroundColor: utils.colors.gray200,
+    borderRadius: utils.borders.radii.basic,
+    paddingLeft: 14,
+    paddingRight: 14.25,
+    extend: [
+        {
+            condition: isFocused,
+            style: {
+                borderColor: utils.colors.gray500,
+            },
+        },
+    ],
+}));
+
+export const SearchInput = ({
+    value,
+    onChange,
+    placeholder,
+    isClearable = false,
+    isDisabled = false,
+}: InputProps) => {
+    const { applyStyle } = useNativeStyles();
+    const [isFocused, setIsFocused] = useState<boolean>(false);
+
+    const handleClear = () => {
+        // Wanted behaviour for clearing text from input.
+        onChange('');
+    };
+
+    return (
+        <Box style={applyStyle(inputWrapperStyle, { isFocused })}>
+            <View>
+                <Icon type="search" color="gray600" />
+            </View>
+            <TextInput
+                value={value}
+                onChangeText={onChange}
+                placeholder={placeholder}
+                maxLength={MAX_LENGTH}
+                editable={!isDisabled}
+                onFocus={() => setIsFocused(true)}
```

Comment:
use `useCallback`

Reply:
I kind of fan where you use it everywhere, because I never seen any performance degradation because of overusing it, but I have seen lot of performance problems because it was not used.

For now let's keep it simple here and don't do that for these simple components and we will see if it will affect performance.

Reply:
Yeah I think the performance optimization is just way too small to even be noticed. I just added the article to explain my thought process behind choosing not to use the `useCallback` hook. Should we just use it for everything then? Because the performance hit will come in much worse in cases where it's actually helpful and is not used at all.

Reply:
I heard other people using it everywhere too. Honestly, I can't say what is better: one is easier but bight cause problems the other one is more perfectionist and performant but requires brain work :D Might be a topic for code improvements + let's scour some open source React repos and see what they are up to :]

Reply:
@dahaca Do you agree I just leave it like this so it doesn't block this PR and then if we decide to have it consistently throughout the whole repo, I will take care of refactoring it? So far it's less then 10 files and it's just moving the position of types.

Reply:
Yes! I unresolved the thread just for visibility :)